### ISSUE TITLE
Web development and C# (and Csla)

### ISSUE THREAD

**Issue by Chicagoan2016** [MarimerLLC/cslaforum#854]
Hi all,
We have invested a huge number of man-hours in some asp.net web forms applications over the last 10-12 years. Recently I was asked to help with one ASP.NET web form application which was relatively simple, we had used Csla in our business layer. The recent changes required opening 'dialog' windows based on some business logic. Since this was an ASP.NET webform application I decided to show those 'dialog' windows from code behind based on broken business rules.
We have been using jQuery in favor of ASP.NET ajax for a long time now.
What I end up doing is ,controlling the work flow of the application essentially from jQuery and C# (and hence Csla )took a  back seat (which made the whole application a 'mess' IMHO)

What should we do for future applications? Any ideas since we can't open 'dialog' windows from C# code without constantly calling methods like RegisterStartupScript() (which I really dont like).

Kind Regards

**Version and Platform**
CSLA version: 4.7.100
OS: Windows,
Platform: ASP.NET

**Comment by rockfordlhotka** [comment]
I must confess that I haven't done any work with Web Forms for probably over a decade. Hopefully someone with more recent experience has some ideas.

**Comment by Chicagoan2016** [comment]
@rockfordlhotka , thank you for replying, do you think if we move to Asp.NET MVC for future projects, will it help us with the scenario I described-- where my jQuery code is essentially the 'driver'?

Regards

**Comment by rockfordlhotka** [comment]
Most web apps today seem to be created using Angular or React, and so are smart client apps that interact with services hosted by the server. Those UI frameworks have support for dialog windows and many other concepts familiar to smart client developers.

There's no doubt that you can have MVC Razor pages push out JavaScript to display dialog windows.

Really I think this is an architectural decision for you. Are you writing a server-side web site that uses bits of JavaScript to do fancy stuff in the browser? Or are you writing a smart client app that is deployed from the web server, and calls back to the web server for data?

**Comment by Chicagoan2016** [comment]
@rockfordlhotka , I agree that scenarios like I mentioned above require different architecture.
Considering the two architectures you mentioned, is my understanding correct?

1) Server side apps that use java script --- For these we could use ASP.NET MVC along with Csla?
2) Smart client apps -- We will use Angular or React, but does it mean we can't use Csla and ASP.NET MVC?

kind regards

**Comment by rockfordlhotka** [comment]
That is correct, you can use CSLA in ASP.NET (MVC or Razor Pages - there are current samples for both). 

Because CSLA can only work where there is .NET, you can't run CSLA client-side in an Angular or React app because they are probably written using TypeScript.

This is where Blazor is so compelling. Blazor allows you to run .NET in the browser (or on the server), and so _all_ the benefits of CSLA come into play.
